### PR TITLE
[PropertyInfo] Handle collection in PhpStan same as PhpDoc

### PR DESCRIPTION
**Pull request by mtarld**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Issues        | Fix https://github.com/symfony/symfony/pull/52699#issuecomment-2199871592
| License       | MIT

On https://github.com/symfony/symfony/pull/52699, a [comment](https://github.com/symfony/symfony/pull/52699#discussion_r1553345349) suggested to support classes that inherit from collection classes (such as doctrine collections for example).
This has been done for PHPDoc, but not in PHPStan.

This PR adds this missing behavior to PHPStan (and adds the missing PHPDoc related test).

### CODE REVIEW

**Review by xabbuh — APPROVED** [review]
I am not completely convinced that allowing subclasses in #52699 was a good decision as it rather looks like introducing a new feature in a patch release. But now that we did it for the `PhpDocTypeHelper` updating the `PhpStanTypeHelper` in the same way looks reasonable to me.

File: src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 namespace Symfony\Component\PropertyInfo\Tests\Extractor;
 
+use Symfony\Component\PropertyInfo\Tests\Fixtures\DummyCollection;
```
Comment (chalasr) [inline src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php]: wrong order, fabbot seems KO

**Review by chalasr — APPROVED** [review]
once CS fixed

**Comment by xabbuh** [comment]
Thank you @mtarld.

**Pull request merged**